File: fix-gameloop.js
const fs = require('fs');
const filePath = 'c:\\Users\\fabia\\Desktop\\STARFORCE-Dark-Horizon\\script\\gameLoop.js';
const buf = fs.readFileSync(filePath);
const useCRLF = buf.includes(Buffer.from('\r\n'));
const sep = useCRLF ? '\r\n' : '\n';
const content = buf.toString('utf8');
const lines = content.split(sep);

console.log('Lines 99-110 (0-indexed 98-109):');
for (let i = 98; i < 110; i++) {
  console.log(i+1, JSON.stringify(lines[i]));
}

// Lines to remove: 103,104,105,106 (1-indexed) = indices 102,103,104,105 (0-indexed)
// Verify they are: orphaned separator + blank lines
const removed = lines.splice(102, 4);
console.log('Removed lines:', removed.map(l => JSON.stringify(l)));

console.log('Lines 99-110 after fix:');
for (let i = 98; i < 108; i++) {
  console.log(i+1, JSON.stringify(lines[i]));
}

const newContent = lines.join(sep);
fs.writeFileSync(filePath, newContent, 'utf8');
console.log('File written successfully.');

File: fix_gameloop.py
import re

file_path = r'c:\Users\fabia\Desktop\STARFORCE-Dark-Horizon\script\gameLoop.js'

# Read the file
with open(file_path, 'r', encoding='utf8') as f:
    content = f.read()

# Try to find and display the problematic section around line 110
lines = content.split('\n')
print("Lines 102-116:")
for i in range(101, 116):
    if i < len(lines):
        line = lines[i]
        # Show with visible whitespace
        visible = line.replace(' ', '·').replace('\t', '→')
        print(f"{i+1}: [{visible}]")

# Pattern to match the garbage block with flexible whitespace
pattern = r'    // -+\s+\n\s*\n(\s*\n)+\s*// -+\s*\n\s*// PLAYER & SPAWN\s*\n\s*// -+\s*\n\s*// -+\s*\n\s*// WINGMAN'

replacement = r'    // -----------------------------------\n    // WINGMAN'

new_content = re.sub(pattern, replacement, content)

if new_content != content:
    print("\n✓ Pattern matched! Writing fix...")
    with open(file_path, 'w', encoding='utf8') as f:
        f.write(new_content)
    
    # Verify
    with open(file_path, 'r', encoding='utf8') as f:
        lines = f.read().split('\n')
    print("\nFixed section (lines 98-120):")
    for i in range(97, 120):
        if i < len(lines):
            print(f"{i+1}: {lines[i]}")
else:
    print("\n✗ Pattern did not match")
